src/test/java/com/puppycrawl/tools/checkstyle/api/DetailASTTest.java

```diff
@@ -143,15 +143,30 @@ private static void checkFile(String filename) throws Exception {
         final FileContents contents = new FileContents(text);
         final DetailAST rootAST = TreeWalker.parse(contents);
         if (rootAST != null) {
-            checkTree(rootAST, null, null, filename, rootAST);
+            checkTree(filename, rootAST);
         }
     }
 
-    private static void checkTree(final DetailAST node,
-                           final DetailAST parent,
-                           final DetailAST prev,
-                           final String filename,
-                           final DetailAST root) {
+    private static void checkTree(final String filename, final DetailAST root) {
+        DetailAST curNode = root;
+        while (curNode != null) {
+            checkNode(curNode, curNode.getParent(), curNode.getPreviousSibling(), filename, root);
```

Comment:
This test also assumes `getFirstChild` and `getNextSibling` are working correct. We can assume `getParent` and `getPreviousSibling` are valid after this `check` call.